packages/flutter/test/rendering/editable_test.dart

```diff
@@ -955,6 +955,67 @@ void main() {
     expect(delegate.textEditingValue.text, '');
   }, skip: isBrowser); // https://github.com/flutter/flutter/issues/61021
 
+  test('arrow keys work after detaching the widget and attaching it again', () async {
+    final TextSelectionDelegate delegate = FakeEditableTextState()
+      ..textEditingValue = const TextEditingValue(
+          text: 'W Szczebrzeszynie chrząszcz brzmi w trzcinie',
+          selection: TextSelection.collapsed(offset: 0),
+        );
+    final ViewportOffset viewportOffset = ViewportOffset.zero();
+    final RenderEditable editable = RenderEditable(
+      backgroundCursorColor: Colors.grey,
+      selectionColor: Colors.black,
+      textDirection: TextDirection.ltr,
+      cursorColor: Colors.red,
+      offset: viewportOffset,
+      textSelectionDelegate: delegate,
+      onSelectionChanged: (TextSelection selection, RenderEditable renderObject, SelectionChangedCause cause) {
+        renderObject.selection = selection;
+      },
+      startHandleLayerLink: LayerLink(),
+      endHandleLayerLink: LayerLink(),
+      text: const TextSpan(
+        text: 'W Szczebrzeszynie chrząszcz brzmi w trzcinie',
+        style: TextStyle(
+          height: 1.0, fontSize: 10.0, fontFamily: 'Ahem',
+        ),
+      ),
+      selection: const TextSelection.collapsed(
+        offset: 0,
+      ),
+    );
+
+    final PipelineOwner pipelineOwner = PipelineOwner();
+    editable.attach(pipelineOwner);
+    editable.hasFocus = true;
+    editable.detach();
+    layout(editable);
+    editable.hasFocus = true;
+    editable.selectPositionAt(from: Offset.zero, cause: SelectionChangedCause.tap);
```

Comment:
nit: are line 994 and line 995 essentially doing the same thing (setting the selection to `(0, 0)`)?

Reply:
This was pretty much copied from other tests.